loader/src/main/kotlin/net/weavemc/loader/mixin/MixinSandbox.kt

```diff
@@ -297,14 +314,25 @@ private fun <R> counter() = object : ReadOnlyProperty<R, Int> {
  */
 // TODO: generalize
 class LoaderClassWriter(
-    private val loader: ClassLoader,
+    private val loader: ClassLoader?,
     reader: ClassReader? = null,
     flags: Int = 0,
     private val useBytecodeInheritance: Boolean = true,
 ) : ClassWriter(reader, flags) {
-    override fun getClassLoader(): ClassLoader = loader
-    private fun String.load() = loader.getResourceAsStream("$this.class")?.readBytes()?.asClassReader()?.asClassNode()
-    private val ClassNode.isInterface: Boolean get() = access and ACC_INTERFACE != 0
+    private fun getStream(name: String) = loader?.getResourceAsStream(name)
+        ?: LoaderClassWriter::class.java.getResourceAsStream(name)
+
+    private fun String.load() =
+        getStream("$this.class")?.readBytes()?.asClassReader()?.asClassNode()
+
+    private val ClassNode.isInterface: Boolean
+        get() = access and ACC_INTERFACE != 0
+
+    init {
+        if (!useBytecodeInheritance && loader == null) {
+            error("Cannot use reflection for inheritance without a ClassLoader")
+        }
```

Comment:
fwiw, `null` represents the bootstrap classloader, we might want to support that, not important for this pr probably though.